apps/wallet-mobile/src/kernel/navigation.tsx

```diff
@@ -650,66 +649,3 @@ const getFocusedRouteName = (state: Partial<NavigationState> | NavigationState['
 
   return [name]
 }
-
-function useKeepRoutesInHistory(routesToKeep: string[]) {
-  const navigation = useNavigation()
-  const [initialRouteId] = React.useState(() => getNavigationRouteId(navigation))
-
-  React.useEffect(() => {
-    const currentRouteId = getNavigationRouteId(navigation)
-
-    if (currentRouteId !== initialRouteId) {
-      return
-    }
-    const {routes} = navigation.getState()
-    const currentRouteNames = routes.map((r) => r.name)
-
-    if (compareArrays(currentRouteNames, routesToKeep)) {
-      return
-    }
-
-    const newRoutes = routes.filter((r) => routesToKeep.includes(r.name))
-
-    const task = InteractionManager.runAfterInteractions(() => {
-      const newState = {
-        index: newRoutes.length - 1,
-        routes: newRoutes.map((r) => ({...r, state: undefined})),
-        routeNames: newRoutes.map((r) => r.name),
-      }
-      navigation.reset(newState)
-    })
-
-    return () => task.cancel()
-  }, [navigation, initialRouteId, routesToKeep])
-}
-
-export function useOverridePreviousRoute<RouteName extends string>(previousRouteName: RouteName) {
```

Comment:
It was only used precisely for those Send navigation overrides, and it's buggy which caused the issue.
Turns out adjusting the history state in a hook that executes every render, based on a route name that can change between renders just isn't safe.

Reply:
nice catch!